File: FASE1/LAB2/Ejercicio3/app/src/main/java/com/example/ejercicio3/MainActivity.java
package com.example.ejercicio3;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.RadioButton;
import android.widget.TextView;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    EditText editTextCantidad;
    RadioButton radioDolares, radioSoles;
    Button btnConvertir;
    TextView txtResultado;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        editTextCantidad = findViewById(R.id.editTextCantidad);
        radioDolares = findViewById(R.id.radioDolares);
        radioSoles = findViewById(R.id.radioSoles);
        btnConvertir = findViewById(R.id.btnConvertir);
        txtResultado = findViewById(R.id.txtResultado);

        btnConvertir.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                convertir();
            }
        });
    }

    private void convertir() {
        String texto = editTextCantidad.getText().toString().trim();

        if (texto.isEmpty()) {
            Toast.makeText(this, "Ingrese una cantidad", Toast.LENGTH_SHORT).show();
            return;
        }

        double cantidad;
        try {
            cantidad = Double.parseDouble(texto);
        } catch (NumberFormatException e) {
            Toast.makeText(this, "Ingrese un número válido", Toast.LENGTH_SHORT).show();
            return;
        }

        if (cantidad <= 0) {
            Toast.makeText(this, "Ingrese una cantidad mayor a 0", Toast.LENGTH_SHORT).show();
            return;
        }

        double tipoCambio = 3.65;
        double resultado;

        if (radioDolares.isChecked()) {
            resultado = cantidad / tipoCambio;
            txtResultado.setText(
                    String.format("Resultado: %.2f soles = %.2f dólares", cantidad, resultado)
            );

            radioDolares.setChecked(false);
            radioSoles.setChecked(true);

        } else if (radioSoles.isChecked()) {
            resultado = cantidad * tipoCambio;
            txtResultado.setText(
                    String.format("Resultado: %.2f dólares = %.2f soles", cantidad, resultado)
            );

            radioSoles.setChecked(false);
            radioDolares.setChecked(true);

        } else {
            Toast.makeText(this, "Seleccione una opción", Toast.LENGTH_SHORT).show();
        }
    }
}

File: FASE1/LAB2/Ejercicio1/app/src/main/java/com/example/e01_moneda/MainActivity.kt
package com.example.e01_moneda

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import android.widget.ArrayAdapter
import android.widget.Button
import android.widget.EditText
import android.widget.Spinner
import android.widget.TextView
import android.widget.Toast

class MainActivity : AppCompatActivity() {

    private lateinit var editTextCantidad: EditText
    private lateinit var spinnerOrigen: Spinner
    private lateinit var spinnerDestino: Spinner
    private lateinit var btnConvertir: Button
    private lateinit var txtResultado: TextView

    private val monedas = listOf(
        "Nuevo Sol (PEN)",
        "Dólar (USD)",
        "Euro (EUR)",
        "Libra (GBP)",
        "Rupia (INR)",
        "Real (BRL)",
        "Peso Mexicano (MXN)",
        "Yuan (CNY)",
        "Yen (JPY)"
    )

    private val tasasEnPen = mapOf(
        "Nuevo Sol (PEN)" to 1.0,
        "Dólar (USD)" to 3.65,
        "Euro (EUR)" to 3.98,
        "Libra (GBP)" to 4.65,
        "Rupia (INR)" to 0.044,
        "Real (BRL)" to 0.74,
        "Peso Mexicano (MXN)" to 0.21,
        "Yuan (CNY)" to 0.51,
        "Yen (JPY)" to 0.024
    )

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        editTextCantidad = findViewById(R.id.editTextCantidad)
        spinnerOrigen = findViewById(R.id.spinnerOrigen)
        spinnerDestino = findViewById(R.id.spinnerDestino)
        btnConvertir = findViewById(R.id.btnConvertir)
        txtResultado = findViewById(R.id.txtResultado)

        val adapter = ArrayAdapter(this, android.R.layout.simple_spinner_item, monedas)
        adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)

        spinnerOrigen.adapter = adapter
        spinnerDestino.adapter = adapter

        spinnerOrigen.setSelection(0)
        spinnerDestino.setSelection(1)

        btnConvertir.setOnClickListener {
            convertirMoneda()
        }
    }

    private fun convertirMoneda() {
        val cantidadTexto = editTextCantidad.text.toString().trim()

        if (cantidadTexto.isEmpty()) {
            Toast.makeText(this, "Ingresa una cantidad", Toast.LENGTH_LONG).show()
            return
        }

        val cantidad = cantidadTexto.toDoubleOrNull()

        if (cantidad == null || cantidad <= 0) {
            Toast.makeText(this, "Ingresa un número válido mayor a 0", Toast.LENGTH_LONG).show()
            return
        }

        val monedaOrigen = spinnerOrigen.selectedItem.toString()
        val monedaDestino = spinnerDestino.selectedItem.toString()

        if (monedaOrigen == monedaDestino) {
            txtResultado.text = "Resultado: ${"%.2f".format(cantidad)}"
            return
        }

        val tasaOrigen = tasasEnPen[monedaOrigen]
        val tasaDestino = tasasEnPen[monedaDestino]

        if (tasaOrigen == null || tasaDestino == null) {
            Toast.makeText(this, "No se pudo convertir", Toast.LENGTH_LONG).show()
            return
        }

        val cantidadEnPen = cantidad * tasaOrigen
        val resultado = cantidadEnPen / tasaDestino

        txtResultado.text =
            "Resultado: ${"%.2f".format(cantidad)} $monedaOrigen = ${"%.2f".format(resultado)} $monedaDestino"
    }
}